ts/kpt-functions/src/types.ts

```diff
@@ -206,10 +208,38 @@ export class Configs {
     return cm ? cm.get(key) : undefined;
   }
 
+  /**
+   * Detects if an unknown value has been provided to the config map
+   * @param expectedDataKeys The set of keys expected in the Configs data
+   * @returns Returns undefined if the config map is undefined. Otherwise
+   * returns a string[] containing the invalid keys. The string[] will be empty
+   * if all of the config maps keys are members of the expecteKeys.
+   */
+  hasUnexpectedFunctionParameter(
+    expectedDataKeys: string[]
```

Comment:
I guess it's better to make input as a `Set` and it appears that TS support Set natively.